fuel-tx/src/transaction/types/input/message.rs

```diff
@@ -204,216 +197,64 @@ where
     }
 }
 
-impl<Specification> SizedBytes for Message<Specification>
-where
-    Specification: MessageSpecification,
-{
-    #[inline(always)]
-    fn serialized_size(&self) -> usize {
-        let data_size = if let Some(data) = self.data.as_field() {
-            bytes::padded_len(data.as_slice())
-        } else {
-            0
-        };
-        let predicate_size = if let Some(predicate) = self.predicate.as_field() {
-            bytes::padded_len(predicate.as_slice())
-        } else {
-            0
-        };
-        let predicate_date_size =
-            if let Some(predicate_data) = self.predicate_data.as_field() {
-                bytes::padded_len(predicate_data.as_slice())
-            } else {
-                0
-            };
-
-        MessageSizes::LEN + data_size + predicate_size + predicate_date_size
-    }
-}
+impl FullMessage {
+    pub fn into_message_data_signed(self) -> MessageDataSigned {
+        let Self {
+            sender,
+            recipient,
+            amount,
+            nonce,
+            witness_index,
+            data,
+            ..
+        } = self;
 
-#[cfg(feature = "std")]
-impl<Specification> std::io::Read for Message<Specification>
-where
-    Specification: MessageSpecification,
-{
-    fn read(&mut self, full_buf: &mut [u8]) -> std::io::Result<usize> {
-        let serialized_size = self.serialized_size();
-        if full_buf.len() < serialized_size {
-            return Err(bytes::eof())
+        Message {
+            sender,
+            recipient,
+            amount,
+            nonce,
+            witness_index,
+            data,
+            predicate: (),
+            predicate_data: (),
+            predicate_gas_used: (),
```

Comment:
Can these be replaced with `..Default::default()`?

Reply:
79d57179